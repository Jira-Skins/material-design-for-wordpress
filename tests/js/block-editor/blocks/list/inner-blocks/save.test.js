/**
 * External dependencies
 */
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';

/**
 * Internal dependencies
 */
import Save from '../../../../../../assets/src/block-editor/blocks/list/inner-blocks/list-item/save';

/**
 * Shallow render the component.
 *
 * @param {Object} props - Component props.
 * @return {Function} A functional component.
 */
const setup = props => {
	return render( <Save { ...props } /> );
};

const baseProps = {
	attributes: {
		url: 'http://google.com',
		linkTarget: '_blank',
		leadingIcon: {
			hex: 59517,
			name: 'favorite',
		},
		trailingIcon: {
			name: 'account_balance wallet',
			hex: 59472,
		},
		primaryText: 'List Item Title',
		secondaryText: 'List Item Caption',
	},
};

describe( 'Save', () => {
	it( 'matches snapshot', () => {
		const wrapper = setup( { ...baseProps } );
		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'matches snapshot when invalid URL and no linkTarget', () => {
		const wrapper = setup( {
			attributes: {
				...baseProps.attributes,
				url: undefined,
				linkTarget: undefined,
			},
		} );
		expect( wrapper ).toMatchSnapshot();
	} );
} );
