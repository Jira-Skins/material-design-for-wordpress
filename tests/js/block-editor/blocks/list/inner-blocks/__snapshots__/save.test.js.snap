// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Save matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="mdc-list-item list-item"
        tabindex="0"
      >
        <i
          class="mdc-list-item__graphic material-icons"
        >
          
        </i>
        <a
          class="list-item__link"
          href="http://google.com"
          target="_blank"
        >
           
        </a>
        <span
          class="mdc-list-item__text"
        >
          <span
            class="mdc-list-item__primary-text list-item__text"
          >
            List Item Title
          </span>
          <span
            class="mdc-list-item__secondary-text list-item__text"
            contenteditable="false"
          >
            List Item Caption
          </span>
        </span>
        <i
          class="mdc-list-item__meta material-icons"
        >
          
        </i>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="mdc-list-item list-item"
      tabindex="0"
    >
      <i
        class="mdc-list-item__graphic material-icons"
      >
        
      </i>
      <a
        class="list-item__link"
        href="http://google.com"
        target="_blank"
      >
         
      </a>
      <span
        class="mdc-list-item__text"
      >
        <span
          class="mdc-list-item__primary-text list-item__text"
        >
          List Item Title
        </span>
        <span
          class="mdc-list-item__secondary-text list-item__text"
          contenteditable="false"
        >
          List Item Caption
        </span>
      </span>
      <i
        class="mdc-list-item__meta material-icons"
      >
        
      </i>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Save matches snapshot when invalid URL and no linkTarget 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="mdc-list-item list-item"
        tabindex="0"
      >
        <i
          class="mdc-list-item__graphic material-icons"
        >
          
        </i>
        <a
          class="list-item__link"
          href="#"
        >
           
        </a>
        <span
          class="mdc-list-item__text"
        >
          <span
            class="mdc-list-item__primary-text list-item__text"
          >
            List Item Title
          </span>
          <span
            class="mdc-list-item__secondary-text list-item__text"
            contenteditable="false"
          >
            List Item Caption
          </span>
        </span>
        <i
          class="mdc-list-item__meta material-icons"
        >
          
        </i>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="mdc-list-item list-item"
      tabindex="0"
    >
      <i
        class="mdc-list-item__graphic material-icons"
      >
        
      </i>
      <a
        class="list-item__link"
        href="#"
      >
         
      </a>
      <span
        class="mdc-list-item__text"
      >
        <span
          class="mdc-list-item__primary-text list-item__text"
        >
          List Item Title
        </span>
        <span
          class="mdc-list-item__secondary-text list-item__text"
          contenteditable="false"
        >
          List Item Caption
        </span>
      </span>
      <i
        class="mdc-list-item__meta material-icons"
      >
        
      </i>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
